### PR TITLE
#7 Create Explode Method in enemy script. Add explode particle system …

### PR DESCRIPTION
**Pull request by mserier**
…asset

### CODE REVIEW

File: src/Assets/LightSkyDefense/Scripts/Enemy.cs [hunk 1]
```diff
@@ -27,6 +30,7 @@ void Update()
 
         if (health<=0)
```
Comment (hassanAlAwwadi) [inline src/Assets/LightSkyDefense/Scripts/Enemy.cs]: waarom word dit elke frame gecheked in plaats van alleen wanneer de health word geupdate

Comment (mserier) [reply src/Assets/LightSkyDefense/Scripts/Enemy.cs]: Fixed in latest commit

File: src/Assets/LightSkyDefense/Scripts/Enemy.cs [hunk 2]
```diff
@@ -63,7 +65,7 @@ public void Damage(float dmgAmount)
     /// <param name="healAmount"></param>
     public void Heal(float healAmount)
     {
-        if(health+healAmount>100)
```
Comment (leroykorterink) [inline src/Assets/LightSkyDefense/Scripts/Enemy.cs]: We can reduce this to one line by using modulo.

```health = heal + healAmount % maxHealth;```

File: src/Assets/LightSkyDefense/Scripts/Enemy.cs [hunk 3]
```diff
@@ -73,4 +79,23 @@ public void Heal(float healAmount)
...
+        {
+            explodeEffectInstance = Instantiate(explodeEffect, this.transform.position, new Quaternion());
+        }
+        //play effect
+        explodeEffectInstance.Play();
+        //Destroy after particle (emit) duration + maximum particle lifetime
+        Destroy(explodeEffectInstance.gameObject, (explodeEffectInstance.main.duration + explodeEffectInstance.main.startLifetime.constantMax));
+
+        //Kill enemy (if Explode() called when the enemy was still alive)
+        Destroy(this.gameObject);
```
Comment (leroykorterink) [inline src/Assets/LightSkyDefense/Scripts/Enemy.cs]:
```suggestion
        Destroy(gameObject);
```

File: src/Assets/LightSkyDefense/Scripts/Enemy.cs [hunk 4]
```diff
@@ -73,4 +79,23 @@ public void Heal(float healAmount)
...
+    /// <summary>
+    /// This function kills the enemy and starts the explosion particle system
+    /// </summary>
+    public void Explode()
+    {
+        //if not in the world, instantiate
+        //if(explodeEffect.scene.name == null)
+        if (explodeEffectInstance == null)
+        {
+            explodeEffectInstance = Instantiate(explodeEffect, this.transform.position, new Quaternion());
```
Comment (leroykorterink) [inline src/Assets/LightSkyDefense/Scripts/Enemy.cs]:
```suggestion
            explodeEffectInstance = Instantiate(explodeEffect, transform.position, new Quaternion());
```

File: src/Assets/LightSkyDefense/Scripts/Enemy.cs [hunk 5]
```diff
@@ -73,4 +79,23 @@ public void Heal(float healAmount)
         }
     }
 
+    /// <summary>
+    /// This function kills the enemy and starts the explosion particle system
+    /// </summary>
+    public void Explode()
+    {
+        //if not in the world, instantiate
+        //if(explodeEffect.scene.name == null)
```
Comment (leroykorterink) [inline src/Assets/LightSkyDefense/Scripts/Enemy.cs]:
```suggestion
```

File: src/Assets/LightSkyDefense/Scripts/Enemy.cs [hunk 6]
```diff
@@ -7,27 +7,23 @@ public class Enemy : MonoBehaviour
...
-        
+
     }
-    
-    void Update()
+
+    void FixedUpdate()
     {
-        if(energy >0)
+        if (energy > 0)
```
Comment (leroykorterink) [inline src/Assets/LightSkyDefense/Scripts/Enemy.cs]: This `if` should be inverted.

```
if(energy < 0) {
    Damage(30);
}
```

File: src/Assets/LightSkyDefense/Scripts/Enemy.cs [hunk 7]
```diff
@@ -45,16 +41,26 @@ public float GetHealth()
...
+        }
     }
 
-    
+
     /// <summary>
     /// This function will remove the specified dmgAmount from the enemy's health
     /// </summary>
     /// <param name="dmgAmount"></param>
     public void Damage(float dmgAmount)
```
Comment (leroykorterink) [inline src/Assets/LightSkyDefense/Scripts/Enemy.cs]: We shouldn't abbreviate variable names. `dmgAmount` should be `damageAmount`.

**Pull request merged**